src/main/java/sirius/biz/storage/layer2/BlobUriParser.java

```diff
@@ -0,0 +1,153 @@
+ */
+public class BlobUriParser {
+
+    /**
+     * Contains the prefix length ("/dasd/") to cut from an incoming URI
+     */
+    private static final int URI_PREFIX_LENGTH = BlobDispatcher.URI_PREFIX_TRAILED.length();
+
+    /**
+     * Contains the prefix length ("xxl" + "/") to cut from an incoming URI
+     */
+    private static final int LARGE_FILE_MARKER_LENGTH = BlobDispatcher.LARGE_FILE_MARKER.length();
+
+    private static final int TYPE = 0;
+    private static final int STORAGE_SPACE = 1;
+    private static final int ACCESS_TOKEN = 2;
+
+    private BlobUriParser() {
+    }
+
+    /**
+     * Parses the given URI into a {@link BlobUri}.
+     *
+     * @param uri the URI to parse
+     * @return the parsed URI as {@link BlobUri}
+     */
+    public static Optional<BlobUri> parseBlobUri(String uri) {
+        if (!uri.startsWith(BlobDispatcher.URI_PREFIX_TRAILED)) {
+            return Optional.empty();
+        }
+
+        BlobUri result = new BlobUri();
+
+        // Cut off "/dasd/"...
+        uri = uri.substring(URI_PREFIX_LENGTH);
+
+        // Cut off "xxl/" if present...
+        if (uri.startsWith(BlobDispatcher.LARGE_FILE_MARKER)) {
+            uri = uri.substring(LARGE_FILE_MARKER_LENGTH);
+            result.withLargeFileExpected(true);
+        }
+
+        Values uriParts = Values.of(uri.split("/"));
+
+        if (uriParts.length() < 3) {
+            return Optional.empty();
+        }
+
+        String type = uriParts.at(TYPE).asString();
+
+        result.withCacheable(type.contains(BlobDispatcher.FLAG_CACHEABLE))
+              .withDownload(type.contains(BlobDispatcher.FLAG_DOWNLOAD))
+              .withStorageSpace(uriParts.at(STORAGE_SPACE).asString())
+              .withAccessToken(uriParts.at(ACCESS_TOKEN).asString());
+
+        if (type.contains(BlobDispatcher.FLAG_PHYSICAL)) {
+            return parsePhysicalUri(result, uriParts);
+        }
+
+        if (type.contains(BlobDispatcher.FLAG_VIRTUAL)) {
+            return parseVirtualUri(result, uriParts);
+        }
+
+        return Optional.empty();
+    }
+
+    @Nonnull
+    private static Optional<BlobUri> parsePhysicalUri(BlobUri result, Values uriParts) {
+        // Handles /dasd/p/SPACE/ACCESS_TOKEN/BLOB_KEY/seo--PHYSICAL_KEY.FILE_EXTENSION
+        if (uriParts.length() == 5) {
+            String filename = stripAdditionalText(uriParts.at(4).asString());
+            String physicalKey = Files.getFilenameWithoutExtension(filename);
+
+            return Optional.of(result.withPhysical(true)
+                                     .withBlobKey(uriParts.at(3).asString())
+                                     .withPhysicalKey(physicalKey)
+                                     .withFilename(filename));
+        }
+
+        // Handles /dasd/p[d]/SPACE/ACCESS_TOKEN/BLOB_KEY/PHYSICAL_KEY/seo--FILENAME.FILE_EXTENSION
+        if (uriParts.length() == 6) {
+            String filename = stripAdditionalText(uriParts.at(5).asString());
```

Comment:
Could we introduce consts for the numbers describing which part of the URI they are addressing?